include/real/convenience.hpp

```diff
@@ -33,15 +32,15 @@ void add_digits(std::vector<T1> &number, T1 num_base, T2 digit, T2 digit_base)
     T1 tmp_carry; // to store current carry, getting created by multiplication
     T1 result; // to store value of result
 
-    for(size_t i = n-1; i>=0; --i)
+    for(size_t i = n-1; i >= 0; --i)
```

Comment:
Can we use for each? I don't think need the i for anything. Some ideas of how to foreach on reverse: https://stackoverflow.com/questions/8542591/c11-reverse-range-based-for-loop

Reply:
Ok, I will change that.

Reply:
Completed the suggested changes and added the commit.